frappe/__init__.py

```diff
@@ -426,11 +426,19 @@ def has_website_permission(doctype, ptype="read", doc=None, user=None, verbose=F
 	if not user:
 		user = session.user
 
-	for method in (get_hooks("has_website_permission") or {}).get(doctype, []):
-		if not call(get_attr(method), doc=doc, ptype=ptype, user=user, verbose=verbose):
-			return False
+	hooks = (get_hooks("has_website_permission") or {}).get(doctype, [])
+	if hooks:
+		for method in hooks:
+			result = call(get_attr(method), doc=doc, ptype=ptype, user=user, verbose=verbose)
+			# if even a single permission check is Falsy
+			if not result:
+				return False
 
-	return True
+		# else it is Truthy
+		return True
+
+	else:
+		return False
 
```

Comment:
Non that's what caused the security issue. I.e when there are no hooks defined return false. 

Sent from my phone

> On 27-May-2015, at 5:09, Rushabh Mehta <notifications@github.com> wrote:
> 
> In frappe/__init__.py:
> 
> >  
> > -	return True
> > +		# else it is Truthy
> > +		return True
> > +
> > +	else:
> > +		return False
> >  
> should be return True
> 
> —
> Reply to this email directly or view it on GitHub.
>